core/change.go

```diff
@@ -100,79 +109,133 @@ func (change Change) Write(writer io.Writer) error {
 	return err
 }
 
-func kindFromLabel(config Config, label string) *KindConfig {
-	for _, kindConfig := range config.Kinds {
-		if kindConfig.Label == label {
-			return &kindConfig
-		}
-	}
-
-	panic("label not part of any kind")
+type PromptContext struct {
+	config      Config
+	stdinReader io.ReadCloser
+	kind        *KindConfig
 }
 
 // AskPrompts will ask the user prompts based on the configuration
 // updating the change as prompts are answered.
-func AskPrompts(change *Change, config Config, stdinReader io.ReadCloser) error {
-	var (
-		err  error
-		kind *KindConfig
-	)
-
-	if len(config.Components) > 0 {
-		compPrompt := promptui.Select{
-			Label: "Component",
-			Items: config.Components,
-			Stdin: stdinReader,
-		}
+func (change *Change) AskPrompts(config Config, stdinReader io.ReadCloser) error {
+	ctx := PromptContext{
+		config:      config,
+		stdinReader: stdinReader,
+		kind:        nil,
+	}
 
-		_, change.Component, err = compPrompt.Run()
-		if err != nil {
-			return err
-		}
+	err := change.promptForComponent(&ctx)
+
+	if err != nil {
+		return err
 	}
 
-	if len(config.Kinds) > 0 {
-		kindPrompt := promptui.Select{
-			Label: "Kind",
-			Items: config.Kinds,
-			Stdin: stdinReader,
-		}
+	err = change.promptForKind(&ctx)
 
-		_, change.Kind, err = kindPrompt.Run()
-		if err != nil {
-			return err
-		}
+	if err != nil {
+		return err
+	}
 
-		kind = kindFromLabel(config, change.Kind)
+	err = change.parseKind(&ctx)
+
+	if err != nil {
+		return err
 	}
 
-	if kind == nil || !kind.SkipBody {
-		bodyPrompt := config.Body.CreatePrompt(stdinReader)
-		change.Body, err = bodyPrompt.Run()
+	err = change.promptForBody(&ctx)
 
-		if err != nil {
-			return err
+	if err != nil {
+		return err
+	}
+
+	return change.promptForUserChoices(&ctx)
+}
+
+func (change *Change) promptForComponent(ctx *PromptContext) error {
+	if len(ctx.config.Components) == 0 {
+		return nil
+	}
+
+	compPrompt := promptui.Select{
```

Comment:
Ok here is what I am thinking when it comes to the validation thing. First, every prompt should be validated and every input should be validated the same way. Therefore, we should create a validation function and pass it to the prompt as well as call it manually if the value was already supplied. Something like:
```go
func BuildComponentValidation(...) func (validation args) {
    return func () {
        // whatever
    }
}
```
I realize now I wrote this on component when your task is for body and kind but the same will apply when we do the custom / component changes in another PR.
This would allow us to test the validation logic separately from either the manually passed values or the prompt values.

Reply:
No problem, I can take a crack at this later tonight.

Reply:
I think it looks fine for now, I might come back through later and clean up some naming decisions or flows.